src/amltk/pipeline/node.py

```diff
@@ -468,15 +468,34 @@ def linearized_fidelity(self, value: float) -> dict[str, int | float | Any]:
 
         return prefix_keys(d, f"{self.name}:")
 
-    def iter(self) -> Iterator[Node]:
-        """Iterate the the nodes, including this node.
+    def iter(self, skip_unchosen: bool = False) -> Iterator[Node]:
+        """Recursively iterate through the nodes starting from this node.
+
+        This method traverses the nodes in a depth-first manner, including
+        the current node and its children nodes.
+
+        Args:
+            skip_unchosen (bool): Flag to skip unchosen nodes in Choice nodes.
```

Comment:
I guess your IDE is putting these in automatically? If not, there's no need to for documentation purposes as the documentation engine will automatically document the type from the function signature.

Don't really need to change it but one reason for changing it is that the documentation and the code doesn't go out of sync, i.e. if the signature is changed from something like `bool | None = False`, then the docstring doesn't go out of sync.

---

Side note, I assume running `ruff` on this will complain. It'll ask for `def iter(self, *, skip_unchosen: bool = False) -> Iterator[Node]`, emphasis on the `*` to indicate keyword only.

Main reason being is that from the callers perspective, it dissallows `node.iter(True)`, where someone else reading has no idea what the `True` is doing. Also prevents issues if another boolean argument is added, in which case the user can't accidentally have the wrong thing,

e.g.. imagine the signature was updated to

```python
# Before update
node.iter(False)

# Code is updated
# This silently breaks the user code above
def iter(self, deep: bool = True, skip_unchosen: bool = False):

```

Reply:
I've added unnecessary docstrings exactly because the linter complains.

Reply:
Interseting, is it PyCharm complaining about no type in the docstring or `ruff`? I've never had `ruff` complain to me about missing docstrings before?